orangecanvas/main.py

```diff
@@ -435,6 +436,17 @@ def show_message(message):
     gc.collect()
 
     del app
+
+    if status == 96:
+        log.info('Restarting via exit code 96.')
+        try:
+            p = psutil.Process(os.getpid())
+            for handler in p.open_files() + p.connections():
+                os.close(handler.fd)
+        except Exception as e:  # pylint: disable=broad-except
+            logging.error(e)
+        return os.execv(sys.executable, ['python', sys.argv[0]])
+
```

Comment:
Can you rebase on gh-157 and use `run_after_exit([sys.executable,  sys.argv[0]])` here?

Reply:
Thanks Aleš, you're a legend for looking into this and implementing it.
I've done it, and tested it with https://github.com/biolab/orange3/pull/5071. It works great.